qiskit_machine_learning/utils/loss_functions/loss_functions.py

```diff
@@ -20,24 +20,57 @@
 
 class Loss(ABC):
     """
-    Abstract base class for Loss.
+    Abstract base class for computing Loss.
     """
 
-    def __call__(self, predict, target):
+    def __call__(self, predict: np.ndarray, target: np.ndarray):
+        """
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Returns:
+            a float value of the loss function
+        """
         return self.evaluate(predict, target)
 
     @abstractmethod
-    def evaluate(self, predict, target):
-        """ evaluate """
+    def evaluate(self, predict: np.ndarray, target: np.ndarray):
```

Comment:
I see in the derived classes they document this returning a float and the meaning of the float. In which case I think the signature here should be updated to return a float ie `def evaluate(self, predict: np.ndarray, target: np.ndarray) -> float:` and document here the meaning of that return.

Reply:
updated

Reply:
For both `evaluate` and `gradient` we expect them to return a float right so the caller of a concrete sub-class knows what to expect. Hence in the abstract methods here the signature should show the float return typehint and the Returns in the docstring document the meaning of the float. The docstrings here would then be complete meaning no docstring should be needed on the derived classes - which at present all have a copy/paste of the same Returns text when they should all just simply have no docstring i.e inherit the one from the base class.